lib/install/management/configure.go

```diff
@@ -98,49 +102,150 @@ func (d *Dispatcher) Configure(vch *vm.VirtualMachine, conf *config.VirtualConta
 		if err != nil {
-			// Update error message and fall through to roll back
 			err = errors.Errorf("Failed to grant permissions to ops-user, failure: %s", err)
 		}
 	}
 
+	// compatible with old version's upgrade snapshot name
+	if oldSnapshot != nil && (vm.IsConfigureSnapshot(oldSnapshot, ConfigurePrefix) || vm.IsConfigureSnapshot(oldSnapshot, UpgradePrefix)) {
+		d.deleteSnapshot(&oldSnapshot.Snapshot, oldSnapshot.Name, conf.Name)
+	}
+
+	return nil
+}
+
+// updateInventory updates VCH Folder inventories.
+//
+// If the VCH resides in the VMFolder then a new folder will
+// be created and the VCH and all Containers will be moved into the
+// new folder.  If the VCH already resides in a folder other than
+// the VMFolder, then no action will be taken.
+//
+// vApp support planned: https://github.com/vmware/vic/issues/7670
+func (d *Dispatcher) updateInventory(name string) error {
+	defer trace.End(trace.Begin("", d.op))
+
+	appName := d.appliance.Name()
+
+	d.op.Infof("Updating inventory for %s or %s", name, appName)
+
+	// get the current appliance folder
+	currentFolder, err := d.appliance.Folder(d.op)
 	if err != nil {
-		// Roll back
-		d.op.Errorf("Failed to upgrade: %s", err)
-		d.op.Infof("Rolling back upgrade")
+		d.op.Errorf("Unable to get current folder:  %s", err)
+		return err
+	}
 
-		if rerr := d.rollback(conf, snapshotName, settings); rerr != nil {
-			d.op.Errorf("Failed to revert appliance to snapshot: %s", rerr)
-			return err
-		}
-		d.op.Infof("Appliance is rolled back to old version")
+	// vApp support planned: https://github.com/vmware/vic/issues/7670
+	if currentFolder.Reference() != d.session.VMFolder.Reference() || d.vchPool.Reference().Type == "VirtualApp" {
+		d.op.Debugf("No inventory update completed for %s", name)
+		return nil
+	}
 
-		d.deleteUpgradeImages(ds, settings)
-		d.retryDeleteSnapshotByRef(newSnapshotRef, snapshotName, conf.Name)
+	// concatenate w/vch name so we can identify if failure
+	tempName := fmt.Sprintf("%s-%s", name, uid.New().Truncate().String())
+	// create target folder with random name for VCH Inventory
+	target, err := d.session.VMFolder.CreateFolder(d.op, tempName)
+	if err != nil {
+		d.op.Errorf("Unable to create target folder during inventory update: %s", err)
+		return err
+	}
 
-		// return the error message for upgrade
+	rp := compute.NewResourcePool(d.op, d.session, d.vchPool.Reference())
+	children, err := rp.GetChildrenVMs(d.op)
+	if err != nil {
+		d.op.Errorf("Unable to locate children during inventory update: %s", err)
 		return err
 	}
 
-	// compatible with old version's upgrade snapshot name
-	if oldSnapshot != nil && (vm.IsConfigureSnapshot(oldSnapshot, ConfigurePrefix) || vm.IsConfigureSnapshot(oldSnapshot, UpgradePrefix)) {
-		d.retryDeleteSnapshotByRef(&oldSnapshot.Snapshot, oldSnapshot.Name, conf.Name)
+	var vicVMs []types.ManagedObjectReference
+	for _, c := range children {
+		// This will make a vSphere call for each VM in the RP - this might
```

Comment:
I'm not a fan of `O(n)` here but it's not likely to be an issue. If reworking this in the future then moving to a call to get the combined results makes sense, i.e. not using the property collector `RetrieveOne` method.